Code/src/main/resources/catalog/catalog.xml

```diff
@@ -18,105 +35,584 @@
-            <function name="acos" signature="Floating a => a -> a"></function>
-            <function name="acosh" signature="Floating a => a -> a"></function>
-            <function name="asin" signature="Floating a => a -> a"></function>
-            <function name="asinh" signature="Floating a => a -> a"></function>
-            <function name="atan" signature="Floating a => a -> a"></function>
-            <function name="atanh" signature="Floating a => a -> a"></function>
-            <function name="ceiling" signature="(RealFrac a, Integral b) => a -> b"></function>
-            <function name="cos" signature="Floating a => a -> a"></function>
-            <function name="cosh" signature="Floating a => a -> a"></function>
-            <function name="exp" signature="Floating a => a -> a"></function>
-            <function name="floor" signature="(RealFrac a, Integral b) => a -> b"></function>
-            <function name="log" signature="Floating a => a -> a"></function>
-            <function name="logBase" signature="Floating a => a -> a -> a"></function>
-            <function name="negate" signature="Num a => a -> a"></function>
-            <function name="pred" signature="Enum a => a -> a"></function>
-            <function name="recip" signature="Fractional a => a -> a"></function>
-            <function name="round" signature="(RealFrac a, Integral b) => a -> b"></function>
-            <function name="signum" signature="Num a => a -> a"></function>
-            <function name="sin" signature="Floating a => a -> a"></function>
-            <function name="sinh" signature="Floating a => a -> a"></function>
-            <function name="sqrt" signature="Floating a => a -> a"></function>
-            <function name="succ" signature="Enum a => a -> a"></function>
-            <function name="tan" signature="Floating a => a -> a"></function>
-            <function name="tanh" signature="Floating a => a -> a"></function>
-            <function name="truncate" signature="(RealFrac a, Integral b) => a -> b"></function>
+        <category name="Tuples">
+        	<function name="fst" signature="(a, b) -> a">
+        		Extract the first component of a pair.
+        	</function>
+        	<function name="snd" signature="(a, b) -> b">
+        		Extract the second component of a pair
+        	</function>
+        	<function name="curry" signature="((a, b) -> c) -> (a, b) -> c">
+        		curry converts an uncurried function to a curried function
+        	</function>
+        	<function name="uncurry" signature="(a -> b -> c) -> (a, b) -> c">
+        		uncurry converts a curried function to a function on pairs
+        	</function>
+        </category>
+        <category name="Equality">
+        	<function name="(==)" signature="Eq a => a -> a -> Bool"></function>
+        	<function name="(/=)" signature="Eq a => a -> a -> Bool"></function>
+        </category>
+        <category name="Ordered Equality">
```

Comment:
Yes, my reasoning was that people are unlikely to have the Hoogle categories memorized, and the small fraction who do will probably not have much problems finding the Equality category when they're looking for the Ordered Equality category.